comps/integrations/haystack/src/opea_haystack/embedders/tei/text_embedder.py

```diff
@@ -0,0 +1,144 @@
+
+from .truncate import EmbeddingTruncateMode
+
+_DEFAULT_API_URL = "http://localhost:6000/embed"
+
+
+@component
+class OPEATextEmbedder:
+    """
+    A component for embedding strings using embedding models provided by
+    [OPEA](https://opea.dev).
+
+    For models that differentiate between query and document inputs,
+    this component embeds the input string as a query.
+
+    Usage example:
+    ```python
+    from opea_haystack.embedders.tei import OPEATextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = OPEATextEmbedder(api_url="http://localhost:6000")
+    text_embedder.warm_up()
+
+    print(text_embedder.run({"text": text_to_embed})
+    ```
+    """
+
+    def __init__(
+        self,
+        api_url: str = _DEFAULT_API_URL,
+        prefix: str = "",
+        suffix: str = "",
+        truncate: Optional[Union[EmbeddingTruncateMode, str]] = None,
+    ):
+        """
+        Create an OPEATextEmbedder component.
+
+        :param api_url:
+            Custom API URL for the OPEA.
+        :param prefix:
+            A string to add to the beginning of each text.
+        :param suffix:
+            A string to add to the end of each text.
+        :param truncate:
+            Specifies how inputs longer that the maximum token length should be truncated.
+            If None the behavior is model-dependent, see the official documentation for more information.
+        """
+
+        self.api_url = api_url
+        self.prefix = prefix
+        self.suffix = suffix
+
+        if isinstance(truncate, str):
+            truncate = EmbeddingTruncateMode.from_str(truncate)
+        self.truncate = truncate
+
+        self.backend: Optional[Any] = None
+        self._initialized = False
+
+    def warm_up(self):
+        """
+        Initializes the component.
+        """
+        if self._initialized:
+            return
+
+        model_kwargs = {"input_type": "query"}
+        if self.truncate is not None:
+            model_kwargs["truncate"] = str(self.truncate)
+        self.backend = OPEABackend(
+            api_url=self.api_url,
+            model_kwargs=model_kwargs,
+        )
+
+        self._initialized = True
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            api_url=self.api_url,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            truncate=str(self.truncate) if self.truncate is not None else None,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "OPEATextEmbedder":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+            The deserialized component.
+        """
+        return default_from_dict(cls, data)
+
```

Comment:
Just as in the document embedder, the text embedder here doesn't need to define `from_dict`.
```suggestion
```